File: solutions/1646-kth-missing-positive-number/solution.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int missingNumbers = arr[arr.length-1]-arr.length;
        if(missingNumbers<k){
            missingNumbers = k-missingNumbers;
            return arr[arr.length-1]+missingNumbers;
        }
        else{
            for (int i = 0; i < arr.length; i++) {
                int missingNumberTillNow = arr[i]-i-1;
                if(missingNumberTillNow>=k){
                    missingNumberTillNow = arr[i]-i-1;
                    return arr[i]+k-missingNumberTillNow-1;
                }
            }
        }
        return 0;
    }
}

File: solutions/1646-kth-missing-positive-number/solution.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int l = 0;
        int h = arr.size()-1;
        while(l<=h){
            int m = l+(h-l)/2;
            int left = arr[m]-m-1;
            if(left<k) l = m+1;
            else h = m-1;
        }
        return h+k+1;
    }
};
